### PR TITLE
Consistent split screen

### PR DESCRIPTION
**Pull request by Fantasmos**
Fix for issue: https://github.com/supertuxkart/stk-code/issues/2816

### CODE REVIEW

File: src/modes/world.cpp [hunk 1]
```diff
@@ -210,6 +210,9 @@ void World::init()
                                : race_manager->getKartIdent(i);
         int local_player_id  = race_manager->getKartLocalPlayerId(i);
         int global_player_id = race_manager->getKartGlobalPlayerId(i);
+		//Replace i with local_player_id, see what happens. 
+		//The error is exactly here, the list is ordered by score, and i 
+		//is used for m_index
```
Comment (qwertychouskie) [inline src/modes/world.cpp]: You have tabs instead of spaces here.

File: src/modes/soccer_world.cpp [hunk 2]
```diff
@@ -422,7 +422,7 @@ AbstractKart *SoccerWorld::createKart(const std::string &kart_ident, int index,
     {
     case RaceManager::KT_PLAYER:
         controller = new LocalPlayerController(new_kart,
-                          StateManager::get()->getActivePlayer(local_player_id));
+			              local_player_id);
```
Comment (qwertychouskie) [inline src/modes/soccer_world.cpp]: And here

**Comment by Fantasmos** [comment]
@qwertychouskie  Thanks a ton, both have been fixed

**Comment by auriamg** [comment]
Merged, thanks

**Pull request merged**